lib/matplotlib/figure.py

```diff
@@ -1821,6 +1841,9 @@ def subplot_mosaic(self, mosaic, *, sharex=False, sharey=False,
             The string notation allows only single character Axes labels and
             does not support nesting but is very terse.
 
+            Tuples may be used instead of lists and tuples may not be used
```

Comment:
> Tuples may be used instead of lists

Do we want to allow this explicitly? I'm in favor of having one documented and recommended way, so that usages are consistent and not half of the people are using lists and the other half tuples.
It's a separate topic that tuples are accepted due to duck-typing and can be used in special situations.

Reply:
Agree to drop the first half of this sentence as many things will "work":

```python
import matplotlib.pyplot as plt
from collections import deque

a = deque([deque(['A', 'B']), deque(['C', 'D'])])
plt.subplot_mosaic(a)
plt.show()
```

but we should only document one spelling.  In the tutorial I think we also pass a 2D numpy array.

Maybe the input should be "something coercible to a numpy object array of hashables"?

Reply:
This also "works" 🤦🏻 

```python
plt.subplot_mosaic(frozenset([frozenset('ab'), frozenset('cd')]))
```

Reply:
I updated this wording.

On one hand, no one has complained that they can not use tuples as keys yet, on the other hand if we take "tuple is broadcast" in this PR then relaxing this in the future would cost of some tricky logic around sorting out what to do about:

```python
plt.subplot_mosaic([[('a', 'b'), 'a'], ['b', 'c']], per_subplot_kw={('a', 'b'): {}})
```

I think you have to always treat the tuple as "can I broadcast this" first and then we decide if we fallback to "maybe this is a key?" (as if you do it the other way you can _never_ broadcast to 'a' and 'b' in this case).  We might also be able to break this ambiguity by requiring tuple keys to always be nested in another tuple for `per_subplot_kw` as `{(('a', 'b'),): {}}` which is annoying, but not the worst.

Reply:
I think the wording as you have written it now is the correct solution:

> The Axes identifiers may be `str` or a non-iterable hashable object (e.g. `tuple` s may not be used).